### DISCUSSION TITLE
[Design] New hangout flow - homepage

### DISCUSSION THREAD

**Discussion by lpatmo** [codebuddies/frontend#137] [Ideas]
![image](https://user-images.githubusercontent.com/4512699/90727648-767bd000-e278-11ea-9f1f-ad5a29ad79f8.png)

**Comment by lpatmo** [comment]
A couple of notes:
- a user won’t be asked to enter their timezone and general availability again and again, if they’ve already entered it during account signup
- this idea is similar to the CB Connect flow, but the big difference is that we won’t be matching users.
- Design upcoming for what the “I am looking for ___” list of posts looks like.
- After someone fills out their info, they’ll be able to click a button to start a hangout
- The checkbox is meant to allude to a private session vs. a public session :sweat_smile: Feedback welcome re: how to word it better

**Comment by BethanyG** [comment]
For the session "toggle" how about:

![codebuddies option btn](https://user-images.githubusercontent.com/5923094/90839964-a0330680-e30d-11ea-8e75-9987018613ca.png)

or 

![btn option II](https://user-images.githubusercontent.com/5923094/90839967-a5905100-e30d-11ea-8c31-b4c866c756f5.png)

**Reply by lpatmo** [reply]
LOVE THAT WORDING

... have never thought of using the proper noun CodeBuddies in that particular way, but I kinda like it, heh, since "open to additional code buddies" (lowercase) strikes me as slightly worse

**Comment by lpatmo** [comment]
From @tgrrr: "Maybe some code tags rather than a text area for the I hope to pair on"

**Reply by BethanyG** [reply]
Or a place to put a link to a particular resource/topic!

**Reply by lpatmo** [reply]
Would it be easier to implement search on a freeform text input field for the MVP?

**Reply by BethanyG** [reply]
I was imagining something (really roughly) like:

![tags in hangout flow](https://user-images.githubusercontent.com/5923094/90998289-9cef8300-e578-11ea-9b2b-a50315010a82.png)

**Comment by lpatmo** [comment]
"And multi step forms tend to get more people complete them" - @tgrrr

**Comment by lpatmo** [comment]
"Maybe the city time zone aspect can be moved to an account set up process so they don’t have to enter that in every time" - @angelocordon

**Reply by BethanyG** [reply]
@angelocordon -- hard agree for this.  Also for auto-filling the "sniffed" timezone, with the ability to then change it.....I really worry that a long list of 100+ [timezones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) will chase people away.

**Reply by angelocordon** [reply]
Not specifically directed at this sketch, but perhaps from a user looking for other people to pair with perspective, it would be great to show the pairing events in the current user's timezone.

**Reply by lpatmo** [reply]
💯  Followup question: do we also need to show the original scheduler's timezone? 

On codebuddies.org/hangouts pages right now, we only show the current user's timezone, not what that time translates into for the original poster.

**Reply by angelocordon** [reply]
IMO that's a secondary kind of information (something that you'd hide behind a tooltip or something). I can see it being useful but perhaps not strictly important.

**Reply by angelocordon** [reply]
Just articulating my thoughts here: 

As someone looking for pairs, I want to find pairing events that can match _my_ schedule. Displaying this in their time zone instead of the event owner's time zone would make the user experience smoother as we lessen the mental gymnastics of trying to calculate availability for the user. 

As someone creating pairing events, I would want this in my time zone as well; I know what time I'm available but don't necessarily think about what time zone I'm in or other's timezones. 

Overall, technically speaking: when we save these events, it should be in UTC time (in the database). When presenting events, it should be in the timezone of the person's viewing it. 

Hope that makes sense!

**Reply by lpatmo** [reply]
Cool! What you described is the behavior we currently we have on codebuddies.org/hangouts, so I agree we'd want to recreate it. 

I mainly wanted to confirm whether it'd be helpful for the user to see the timezone from the OP's POV in addition to their own, which is what World Time Buddy has:
![image](https://user-images.githubusercontent.com/4512699/90991775-82a9ab00-e560-11ea-9457-e413015d45c7.png)

**Reply by angelocordon** [reply]
Rad!

Re:
> I mainly wanted to confirm whether it'd be helpful for the user to see the timezone from the OP's POV in addition to their own

🤷🏽‍♂️ Honestly, I don't know. It sounds like not something we need to get out of the gate with; I'd suggest we take note of the idea and implement it if we find a need for it in the future.

**Reply by lpatmo** [reply]
Thanks, that's good feedback 😉  -- I won't include on a priority list.

**Comment by lpatmo** [comment]
"I'm generally available...
Maybe some day of the week checkboxes,
And times could be either morning, afternoon, evening, the wee hours, or other.
Or a slider with times.
Or we could use chips inside of the textbox https://material-ui.com/components/chips/
The benefits of slightly more specific days and times is matching with others" 
- @tgrrr

**Reply by BethanyG** [reply]
Yes - and to tag onto what @angelocordon  has said above, this should be part of the register/profile flow as well, so that users don't have to enter it every time.

**Comment by lpatmo** [comment]
"Btw, might be cool when hangouts are closed, that a form pops up where people can put in their TIL and other feedback. But I would be happy to attend a follow up."

"You could even do this in an email format, or give people the profile option to either have it as a pop-up after the session or as an email, or to opt out." 

- @bengineerdavis

**Reply by angelocordon** [reply]
I'd love this as a transactional email asking "how did your session go?" Maybe not just as a TIL but a way for us to collect feedback and maybe as a way to give feedback to others also (perhaps more anonymously).

**Reply by bengineerdavis** [reply]
It might be good to offer a checkbox that can authorize permission for a username to be supplied or to offer contact info to the dev team if they want to ask them questions. There should also always be an opt-out option. In fact, it would be in the spirit of open source that we always default to an opt-in model, and then send an email out reminding them about this feature, which they can then turn on later.

**Reply by BethanyG** [reply]
The opt-in opt-out **has** to be included for GDPR purposes, as I understand.  Also for CA privacy laws, if I remember correctly.  @billglover - any insights for GDPR?  I'll look at the CA laws.

**Reply by billglover** [reply]
My understanding of GDPR centres on "informed consent" and "user control". I have not read the full regulations. If we wanted to share contact details we'd need to make it clear and receive consent from users that this is what we were going to do. They'd need the ability to withdraw this consent.

> It might be good to offer a checkbox that can authorize permission for a username to be supplied or to offer contact info to the dev team if they want to ask them questions.

Is the aim to ensure that attendees on a hangout can continue to have a follow on conversation once the hangout is complete? When this has happened in the past we've just agreed during the hangout how we'll follow up. This is typically Slack. I have questions though:
* Do we mind where the follow up discussion takes place?
* Are the plans for a discussion feature over and above Slack?
* Do we have thoughts on how users might send messages to each other on CodeBuddies?

In the interests of simplicity, I'd favour leaving this functionality out of an initial release and relying on hangout participants to agree an approach for follow up. As we learn more about how hangouts are used we can add additional features to improve the experience.

**Reply by BethanyG** [reply]
For the CA law ([full text here](http://leginfo.legislature.ca.gov/faces/billNavClient.xhtml?bill_id=201720180AB375)) I think the two biggies are (from a [Vox Media summary article](https://www.vox.com/recode/2019/12/30/21030754/ccpa-2020-california-privacy-law-rights-explained)):

1.  The right of Californians to know what personal information is being collected about them.
     > **What this means:** A business must tell you that it collects personal information about you either before or as that information is collected.

2.  The right of Californians to access their personal information.
      >  **What this means:** Businesses (_and organizations_) must offer you **ways to request a copy of the personal information** they have collected about you, and they must provide it free of charge **within 45 days of your request**. You have the right to know the types of personal information (for example: **name, email, or location)** a business has collected, where it came from, why it was collected, the c**ategories of third parties it has shared that information with**, and the specific pieces of information it has collected. **You can also tell them to delete that information.**

In addition, there is the CAN-SPAM law ([summary from litmus here](https://www.litmus.com/blog/how-to-fix-can-spam-so-it-doesnt-further-harm-u-s-businesses/)) requires that you:
> - Include a working unsubscribe link in every promotional email you send
> - Honor opt-out requests within 10 business days
> - Include a mailing address in every email you send
> - Never use misleading or deceptive sender names, subject lines, or email copy
> - Never attempt to conceal the organizations identity or the fact that you are sending advertising

GDPR is stronger, in that sense, so following GDPR covers CAN-SPAM and the CA Law (_looks like_) as well.

I agree with @billglover -- I think maybe we leave the detailed functionality out of the MVP (_but obviously consider how we build in systems to honor user privacy and the two laws._).
